### DISCUSSION TITLE
docs/models/ipam/asn.md

### DISCUSSION THREAD

**Discussion by DM3740** [netbox-community/netbox#17585] [General]
64-bit AS number support ? Seriously ?
https://www.iana.org/assignments/as-numbers/as-numbers.xhtml

**Comment by candlerb** [comment]
It's a typo in the docs. It should say "16- or 32-bit ASNs"

It's correct in the source code:

```
class ASN(PrimaryModel):
    """
    ...
    asn = ASNField(
        unique=True,
        verbose_name=_('ASN'),
        help_text=_('16- or 32-bit autonomous system number')
    )
```

```
# BGP ASN bounds
BGP_ASN_MIN = 1
BGP_ASN_MAX = 2**32 - 1
...
class ASNField(models.BigIntegerField):
    description = "32-bit ASN field"
    default_validators = [
        MinValueValidator(BGP_ASN_MIN),
        MaxValueValidator(BGP_ASN_MAX),
    ]
```

**Reply by DM3740** [reply]
I opened Netbox and docs yesterday and it seems to me nobody read the docs last two years :-)
"Glory to the programmers who ignore the physical meaning of the calculated values!"

**Comment by jeremystretch** [comment]
What did you hope to achieve with this comment? Why didn't you instead [submit an issue](https://github.com/netbox-community/netbox/issues/new?assignees=&labels=type%3A+documentation%2Cstatus%3A+needs+triage&projects=&template=03-documentation_change.yaml) to correct the documentation?

**Reply by DM3740** [reply]
First day on Github - not all ways explored :-(
